types/IMBuildings/Verticount/uplink.js

```diff
@@ -0,0 +1,173 @@
+          emit("sample", { data: totalCounter, topic: "totalCounter" });
+          break;
+        case 0x08:
+          lifecycle.deviceStatus = bytes[bytes.length - 4];
+          lifecycle.batteryVoltage =
+            readUInt16BE(bytes, bytes.length - 3) / 100;
+          lifecycle.sensorStatus = bytes[bytes.length - 1];
+          break;
+        case 0x09:
+          lifecycle.deviceStatus = bytes[bytes.length - 15];
+          lifecycle.batteryVoltage = readUInt16BE(bytes, bytes.length - 14) / 100;
+          data.counterA = readUInt16BE(bytes, bytes.length - 12);
+          data.counterB = readUInt16BE(bytes, bytes.length - 10);
+          lifecycle.sensorStatus = bytes[bytes.length - 8];
+          totalCounter.totalCounterA = readUInt16BE(bytes, bytes.length - 7);
+          totalCounter.totalCounterB = readUInt16BE(bytes, bytes.length - 5);
+          lifecycle.payloadCounter = bytes[bytes.length - 3];
+          lifecycle.rssi = bytes[bytes.length - 2];
+          lifecycle.ceLevel = bytes[bytes.length - 1];
+          emit("sample", { data, topic: "default" });
+          emit("sample", { data: totalCounter, topic: "totalCounter" });
+          break;
+        case 0x0C:
+          lifecycle.deviceStatus = bytes[bytes.length - 8];
+          lifecycle.batteryVoltage = readUInt16BE(bytes, bytes.length - 7) / 100;
+          lifecycle.sensorStatus = bytes[bytes.length - 5];
+          lifecycle.sensorActivityON = readUInt16BE(bytes, bytes.length - 4);
+          lifecycle.sensorActivityOFF = readUInt16BE(bytes, bytes.length - 2);
+          break;
+        default:
+          break;
+      }
+      break;
+    }
+    case payloadTypes.BUTTONS: {
+      switch (payloadVariant) {
+        case 0x03:
+          lifecycle.deviceStatus = bytes[bytes.length - 4];
+          lifecycle.batteryVoltage =
+            readUInt16BE(bytes, bytes.length - 3) / 100;
+          data.buttonPressed = bytes[bytes.length - 1] !== 0;
+
+          data.great = bytes[bytes.length - 1] & 0x01;
+          data.good = bytes[bytes.length - 1] & 0x02;
+          data.mid = bytes[bytes.length - 1] & 0x04;
+          data.bad = bytes[bytes.length - 1] & 0x08;
+          data.worst = bytes[bytes.length - 1] & 0x10;
+
+          break;
+        case 0x04:
+          lifecycle.deviceStatus = bytes[bytes.length - 13];
+          lifecycle.batteryVoltage =
+            readUInt16BE(bytes, bytes.length - 12) / 100;
+
+          data.great = readUInt16BE(bytes, bytes.length - 10);
+          data.good = readUInt16BE(bytes, bytes.length - 8);
+          data.mid = readUInt16BE(bytes, bytes.length - 6);
+          data.bad = readUInt16BE(bytes, bytes.length - 4);
+          data.worst = readUInt16BE(bytes, bytes.length - 2);
+          break;
+        default:
+          break;
+      }
+      emit("sample", { data, topic: "default" });
+      break;
+    }
+    default:
+      emit("sample", {
+        data: { reason: "UNSUPORTED_PAYLOAD" },
```

Comment:
possibly also emit the payload in this case

Reply:
Added an error topic for all IMBuilding devices